modules/select2_publish/js/select2.publish.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @file
+ * Select2 integration.
+ */
+(function ($, drupalSettings) {
+  'use strict';
+
+  Drupal.behaviors.select2_publish = {
+    attach: function (context) {
+      $('.select2-widget', context).on('select2-init', function (e) {
+        var config = $(e.target).data('select2-config');
+        config.createTag = function (params) {
```

Comment:
I'm not sure about this overwrite of `createTag`. If we would have new sub-module that adds new property to element, it would not work. Maybe we can just use parent function to provide result and then just append what is relevant for this sub-module. For example:
```
        var parentCreateTagHandler = config.createTag;
        config.createTag = function (params) {
          var elem = parentCreateTagHandler(params);

          if (elem) {
            elem.published = $(e.target).data('select2-publish-default')
          }

          return elem;
        };
```